lib/addons/vpc-cni/index.ts

```diff
@@ -1,15 +1,241 @@
-import { CoreAddOn } from "../core-addon";
+import { ClusterInfo, Values } from "../../spi";
+import { CoreAddOn, CoreAddOnProps } from "../core-addon";
+import { loadYaml, readYamlDocument } from "../../utils";
+import { Construct } from 'constructs';
+import { KubectlProvider, ManifestDeployment } from "../helm-addon/kubectl-provider";
 
 /**
- * Implementation of VpcCni EKS add-on.
+ * User provided option for the Helm Chart
+ */
+export interface VpcCniAddOnProps extends CoreAddOnProps {
```

Comment:
Design-wise:

What are all the properties that are needed for ENIConfig (e.g subnets, azs)? Let's move them out to a separate type, such as `CustomNetworkingProps` (or if you have a better name please use that)
I assume fields like subnet Ids, etc. will go there.

In VpcCniAddOnProps we can add one optional field:

```typescript
customNetworkingConfig?: CustomNetworkingConfig
```

In that custom network config, subnets will need to be mandatory I assume, unless you see a use case to create ENIConfig with no subnets.